### PR TITLE
added tip about placeholder format and errors

### PR DESCRIPTION
**Pull request by csobier**
For [Docs issue#592](https://github.com/CartoDB/docs/issues/592).  This is my new attempt at clarifying the placholders format. There are tons of examples (outside of the Named Maps doc, repo/examples folder, tutorials, bl.ocks pages, stack exchange, etc.) that all use different formats for placeholders. As a solution, I added a note and a "Placeholder Errors" section to the existing doc.

@iriberri , @rochoa - does this compromise work for you? I think it satisfies the original issue of the novice developer not understanding the placeholder code, while keeping all of the examples intact.

### CODE REVIEW

**Comment by iriberri** [comment]
My ideal solution would be to have a standard way of writing the placeholders/variables in all the documentation and be consistent with it. This way there's no option to confusion and we don't need to excuse ourselves for not being consistent.

And this is totally outside of the Windshaft scope, but if we are so concerned with examples readability we could add extra love to the CSS code in the examples and implement some kind of highlighting for those places where variables are meant to be, so that it's clear visually.

**Comment by csobier** [comment]
Agreed about consistency. Regarding the bigger issue about examples readability- we discussed that in the last [PR](https://github.com/CartoDB/Windshaft-cartodb/pull/348) and it did not seem like any code enhancements would be made to support this. Hence the reason for this second attempt in the docs.  

I can create consistent examples in the Named Map docs alone, but that doesn't help with rest of the examples out there (from all the above resources) that use different symbols.  Your call. Alternatively, I can just close the reported issue as won't fix.

**Comment by csobier** [comment]
@iriberri , can you review? As discussed, all placeholder request show { } in the Maps API doc examples. We should enforce this standard going forward, for consistency.
This should help the user visualize the difference between placeholder variables and the JSON Response format.  Also made [related changes](https://github.com/CartoDB/docs/pull/749) in the [MapConfig doc and the Named Map tutorial.

File: docs/anonymous_maps.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ cdn_url | URLs to fetch the data using the best CDN for your zone.
 #### Call
 
 ```bash
-curl 'https://documentation.cartodb.com/api/v1/map' -H 'Content-Type: application/json' -d @mapconfig.json
+curl 'https://{documentation}.cartodb.com/api/v1/map' -H 'Content-Type: application/json' -d @mapconfig.json
```
Comment (iriberri) [inline docs/anonymous_maps.md]: `{username}` not documentation!

File: docs/anonymous_maps.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ When you have a layergroup, there are several resources for retrieving layergoup
 These tiles will get just the Mapnik layers. To get individual layers, see the following section.
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/{z}/{x}/{y}.png
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/{z}/{x}/{y}.png
```
Comment (iriberri) [inline docs/anonymous_maps.md]: {username}

File: docs/anonymous_maps.md [hunk 3]
```diff
@@ -89,21 +89,21 @@ The MapConfig specification holds the layers definition in a 0-based index. Laye
 Individual layers can be accessed using that 0-based index. For UTF grid tiles:
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/:layer/{z}/{x}/{y}.grid.json
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/{layer}/{z}/{x}/{y}.grid.json
```
Comment (iriberri) [inline docs/anonymous_maps.md]: {username}, and {layergroupid} for `{c01a54877c62831bb51720263f91fb33:0}`

File: docs/anonymous_maps.md [hunk 4]
```diff
@@ -89,21 +89,21 @@ The MapConfig specification holds the layers definition in a 0-based index. Laye
...
 ```
 
-In this case, `:layer` as 0 returns the UTF grid tiles/attributes for layer 0, the only layer in the example MapConfig.
+In this case, `{layer}` as 0 returns the UTF grid tiles/attributes for layer 0, the only layer in the example MapConfig.
```
Comment (iriberri) [inline docs/anonymous_maps.md]: `layer`, mentioning it as a parameter, without curly brackets

File: docs/anonymous_maps.md [hunk 5]
```diff
@@ -89,21 +89,21 @@ The MapConfig specification holds the layers definition in a 0-based index. Laye
...
 
 If the MapConfig had a Torque layer at index 1 it could be possible to request it with:
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/1/{z}/{x}/{y}.torque.json
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/1/{z}/{x}/{y}.torque.json
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username and layergroupid

File: docs/anonymous_maps.md [hunk 6]
```diff
@@ -89,21 +89,21 @@ The MapConfig specification holds the layers definition in a 0-based index. Laye
...
 ```
 
 #### Attributes defined in `attributes` section
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/:layer/attributes/:feature_id
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/{layer}/attributes/{feature_id}
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username and layergroupid

File: docs/anonymous_maps.md [hunk 7]
```diff
@@ -115,19 +115,19 @@ Which returns JSON with the attributes defined, like:
 #### Blending and layer selection
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/:layer_filter/{z}/{x}/{y}.png
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/{layer_filter}/{z}/{x}/{y}.png
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username and layergroupid

File: docs/anonymous_maps.md [hunk 8]
```diff
@@ -115,19 +115,19 @@ Which returns JSON with the attributes defined, like:
...
 ```
 
 Note: currently format is limited to `png`.
 
-`:layer_filter` can be used to select some layers to be rendered together. `:layer_filter` supports two formats:
+`{layer_filter]` can be used to select some layers to be rendered together. `{layer_filter}` supports two formats:
```
Comment (iriberri) [inline docs/anonymous_maps.md]: `layer_filter` only, as here it stands for the name of a parameter

File: docs/anonymous_maps.md [hunk 9]
```diff
@@ -115,19 +115,19 @@ Which returns JSON with the attributes defined, like:
...
 
 - `all` alias
 
-Using `all` as `:layer_filter` will blend all layers in the layergroup
+Using `all` as `{layer_filter}` will blend all layers in the layergroup
```
Comment (iriberri) [inline docs/anonymous_maps.md]: `layer_filter`, name of parameter

File: docs/anonymous_maps.md [hunk 10]
```diff
@@ -115,19 +115,19 @@ Which returns JSON with the attributes defined, like:
...
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/all/{z}/{x}/{y}.png
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/all/{z}/{x}/{y}.png
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username and layergroupid

File: docs/anonymous_maps.md [hunk 11]
```diff
@@ -135,7 +135,7 @@ https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/
 A list of comma separated layer indexes can be used to just render a subset of layers. For example `0,3,4` will filter and blend layers with indexes 0, 3, and 4.
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/0,3,4/{z}/{x}/{y}.png
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/0,3,4/{z}/{x}/{y}.png
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username and layergroupid

File: docs/anonymous_maps.md [hunk 12]
```diff
@@ -172,7 +172,7 @@ callback | JSON callback name.
 #### Call
 
 ```bash
-curl "https://documentation.cartodb.com/api/v1/map?callback=callback&config=%7B%22version%22%3A%221.0.1%22%2C%22layers%22%3A%5B%7B%22type%22%3A%22cartodb%22%2C%22options%22%3A%7B%22sql%22%3A%22select+%2A+from+european_countries_e%22%2C%22cartocss%22%3A%22%23european_countries_e%7B+polygon-fill%3A+%23FF6600%3B+%7D%22%2C%22cartocss_version%22%3A%222.3.0%22%2C%22interactivity%22%3A%5B%22cartodb_id%22%5D%7D%7D%5D%7D"
+curl "https://{documentation}.cartodb.com/api/v1/map?callback=callback&config={%7B%22version%22%3A%221.0.1%22%2C%22layers%22%3A%5B%7B%22type%22%3A%22cartodb%22%2C%22options%22%3A%7B%22sql%22%3A%22select+%2A+from+european_countries_e%22%2C%22cartocss%22%3A%22%23european_countries_e%7B+polygon-fill%3A+%23FF6600%3B+%7D%22%2C%22cartocss_version%22%3A%222.3.0%22%2C%22interactivity%22%3A%5B%22cartodb_id%22%5D%7D%7D%5D%7D}"
```
Comment (iriberri) [inline docs/anonymous_maps.md]: username

File: docs/named_maps.md [hunk 13]
```diff
@@ -164,10 +164,17 @@ The response back from the API provides the name of your MapConfig as a template
 
 ```javascript
 {
-  "template_id":"name",
+  "template_id":"name"
 }
 ```
 
+#### Placeholder Errors
```
Comment (iriberri) [inline docs/named_maps.md]: C'mon, we write awesome documentation, this is not needed :-)

File: docs/named_maps.md [hunk 14]
```diff
@@ -209,7 +216,7 @@ Valid auth token will be needed, if required by the template.
 curl -X POST \
   -H 'Content-Type: application/json' \
   -d @params.json \
-  'https://documentation.cartodb.com/api/v1/map/named/@template_name?auth_token=AUTH_TOKEN'
+  'https://{documentation}.cartodb.com/api/v1/map/named/{template_name}?auth_token={auth_token}'
```
Comment (iriberri) [inline docs/named_maps.md]: username instead of documentation

File: docs/named_maps.md [hunk 15]
```diff
@@ -261,7 +268,7 @@ Updating a Named Map removes all the Named Map instances, so they need to be ini
 curl -X PUT \
   -H 'Content-Type: application/json' \
   -d @template.json \
-  'https://documentation.cartodb.com/api/v1/map/named/:template_name?api_key=APIKEY'
+  'https://{documentation}.cartodb.com/api/v1/map/named/{template_name}?api_key={api_key}'
```
Comment (iriberri) [inline docs/named_maps.md]: username

File: docs/named_maps.md [hunk 16]
```diff
@@ -303,7 +310,7 @@ api_key | is required
 #### Call
 
 ```bash
-curl -X DELETE 'https://documentation.cartodb.com/api/v1/map/named/:template_name?api_key=APIKEY'
+curl -X DELETE 'https://{documentation}.cartodb.com/api/v1/map/named/{template_name}?api_key={api_key}'
```
Comment (iriberri) [inline docs/named_maps.md]: username

File: docs/named_maps.md [hunk 17]
```diff
@@ -337,7 +344,7 @@ api_key | is required
 #### Call
 
 ```bash
-curl -X GET 'https://documentation.cartodb.com/api/v1/map/named?api_key=APIKEY'
+curl -X GET 'https://{documentation}.cartodb.com/api/v1/map/named?api_key={api_key}'
```
Comment (iriberri) [inline docs/named_maps.md]: username

File: docs/named_maps.md [hunk 18]
```diff
@@ -377,7 +384,7 @@ api_key | is required
 #### Call
 
 ```bash
-curl -X GET 'https://documentation.cartodb.com/api/v1/map/named/:template_name?api_key=APIKEY'
+curl -X GET 'https://{documentation}.cartodb.com/api/v1/map/named/{template_name}?api_key={api_key}'
```
Comment (iriberri) [inline docs/named_maps.md]: username

File: docs/named_maps.md [hunk 19]
```diff
@@ -418,7 +425,7 @@ callback | JSON callback name
 #### Call
 
 ```bash
-curl 'https://documentation.cartodb.com/api/v1/map/named/:template_name/jsonp?auth_token=AUTH_TOKEN&callback=callback&config=template_params_json'
+curl 'https://{documentation}.cartodb.com/api/v1/map/named/{template_name}/jsonp?auth_token={auth_token}&callback=callback&config=template_params_json'
```
Comment (iriberri) [inline docs/named_maps.md]: username

File: docs/quickstart.md [hunk 20]
```diff
@@ -22,10 +22,10 @@ $.ajax({
   type: 'POST',
   dataType: 'json',
   contentType: 'application/json',
-  url: 'https://documentation.cartodb.com/api/v1/map',
+  url: 'https://{documentation}.cartodb.com/api/v1/map',
```
Comment (iriberri) [inline docs/quickstart.md]: username

File: docs/quickstart.md [hunk 21]
```diff
@@ -22,10 +22,10 @@ $.ajax({
...
   data: JSON.stringify(mapconfig),
   success: function(data) {
-    var templateUrl = 'https://documentation.cartodb.com/api/v1/map/' + data.layergroupid + '/{z}/{x}/{y}.png'
+    var templateUrl = 'https://{documentation}.cartodb.com/api/v1/map/' + data.layergroupid + '/{z}/{x}/{y}.png'
```
Comment (iriberri) [inline docs/quickstart.md]: username

File: docs/quickstart.md [hunk 22]
```diff
@@ -96,5 +96,5 @@ Note: all `layers` in `metadata` will always have a `type` string and a `meta` d
 You can use the `layergroupid` to instantiate a URL template for accessing tiles on the client. Here we use the `layergroupid` from the example response above in this URL template:
 
 ```bash
-https://documentation.cartodb.com/api/v1/map/c01a54877c62831bb51720263f91fb33:0/{z}/{x}/{y}.png
+https://{documentation}.cartodb.com/api/v1/map/{c01a54877c62831bb51720263f91fb33:0}/{z}/{x}/{y}.png
```
Comment (iriberri) [inline docs/quickstart.md]: username and layergroupid

File: docs/static_maps_api.md [hunk 23]
```diff
@@ -31,33 +31,33 @@ Param | Description
...
-:format | the bounding box in WGS 84 (EPSG:4326), comma separated values for:
+format | the bounding box in WGS 84 (EPSG:4326), comma separated values for:
 --- | ---
 &#124;_ jpg | will have a default quality of 85.
 
 Note: you can see this endpoint as
 
 ```bash
-GET /api/v1/map/static/bbox/:token/:west,:south,:east,:north/:width/:height.:format`
+GET /api/v1/map/static/bbox/{token}/{west,south,east,north}/{width}/{height}.{format}`
```
Comment (iriberri) [inline docs/static_maps_api.md]: {west},{south},{east},{north}

File: docs/static_maps_api.md [hunk 24]
```diff
@@ -161,7 +160,7 @@ After instantiating a map from a CartoDB account:
 #### Call
 
 ```bash
- GET /api/v1/map/static/center/4b615ff367e498e770e7d05e99181873:1420231989550.8699/14/40.71502926732618/-73.96039009094238/600/400.png
+ GET /api/v1/map/static/center/{4b615ff367e498e770e7d05e99181873:1420231989550.8699/14/40.71502926732618/-73.96039009094238/600/400.png}
```
Comment (iriberri) [inline docs/static_maps_api.md]: brackets  this way are not needed. You can provide a explicit example such as:

`/api/v1/map/static/center/4b615ff367e498e770e7d05e99181873:1420231989550.8699/14/40.71502926732618/-73.96039009094238/600/400.png`

or you can provide a parametrized example as:

`/api/v1/map/static/center/{layergroupid}/{z}/{x}/{y}/{width}/{height}.png`

**Comment by csobier** [comment]
Ah, okay. Makes sense. Updates applied!

**Pull request merged**